### DISCUSSION TITLE
Mermaid vs PlantUML

### DISCUSSION THREAD

**Discussion by samuel-bustamante** [Arquisoft/wiq_es04a#24] [Polls]
We are considering two alternatives regarding the diagram generation tool that we will use for the application documentation. On the one hand, PlantUML allows for automatic deployment, which makes documentation deployment easier. On the other hand, Mermaid provides more visual diagrams that are also easier to create. However, its deployment is not automatic.

**Comment by uo288347** [comment]
I choose Mermaid because it also provides an effortless way for automatically visualizing changes

**Reply by PabloGOP** [reply]
Althught I am choosing mermaid myself, there are tools that allow to do so in PlantUML too.

**Comment by alagoconde** [comment]
I personally prefer PlantUML because of that simplicity that gives us in deployment making us save time; I think that Mermaid just makes diagrams prettier and doesn´t have any important advantage. Maybe someone of you knows something I don´t know, so in case you choose Mermaid I will undestand it.

**Comment by PabloGOP** [comment]
I go for Mermaid in order to improve aesthetics of the diagrams. However, I would suggest that as the code is not supported and it would be neccesary to update images in the docu in order to publish the diagrams, we create a directory with the Mermaid sources so it is also part of version control and included in the whole development team workspace scope.

**Comment by and1na** [comment]
While both Mermaid and PlantUML are popular diagramming tools,in my opinion Mermaid could be preferable for a GitHub project because its simplicity and easy integration. Mermaid's syntax is more simple, making it easier for contributors to create and understand diagrams directly in markdown files.
